docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -1,37 +1,137 @@
 [[painless-execute-api]]
 === Painless execute API
+experimental::[]
+
+The Painless execute API runs a script and returns a result.
+
+[[painless-execute-api-request]]
+==== {api-request-title}
+`POST /_scripts/painless/_execute`
+
+[[painless-execute-api-desc]]
+==== {api-description-title}
+Use this API to build and test scripts, such as when defining a script for a
+{ref}/runtime.html[runtime field]. This API requires very few dependencies, and is
+especially useful if you don't have permissions to write documents on a cluster.
+
+The API uses several _contexts_, which control how scripts are executed, what
+variables are available at runtime, and what the return type is.
+
+Each context requires a script, but additional parameters depend on the context
+you're using for that script.
+
+[[painless-execute-api-request-body]]
+==== {api-request-body-title}
+`script`:: (Required, object)
+The Painless script to execute.
++
+.Properties of `script`
+--
+include::../painless-contexts/painless-runtime-fields-context.asciidoc[tag=runtime-field-emit]
+--
+
+`context`:: (Optional, string)
+The context that the script should run in. Defaults to `painless_test` if no
+context is specified.
++
+.Properties of `context`
+[%collapsible%open]
+====
+`painless_test`::
+The default context if no other context is specified.
+
+`filter`::
+Treats scripts as if they were run inside a `script` query.
+
+`score`::
+Treats scripts as if they were run inside a `script_score` function in a
+`function_score` query.
 
-experimental[The painless execute api is new and the request / response format may change in a breaking way in the future]
-
-The Painless execute API allows an arbitrary script to be executed and a result to be returned.
-
-[[painless-execute-api-parameters]]
-.Parameters
-[options="header"]
-|======
-| Name              | Required  | Default                | Description
-| `script`          | yes       | -                      | The script to execute.
-| `context`         | no        | `painless_test`        | The context the script should be executed in.
-| `context_setup`   | no        | -                      | Additional parameters to the context.
-|======
-
-==== Contexts
-
-Contexts control how scripts are executed, what variables are available at runtime and what the return type is.
-
-===== Painless test context
-
-The `painless_test` context executes scripts as is and does not add any special parameters.
-The only variable that is available is `params`, which can be used to access user defined values.
-The result of the script is always converted to a string.
-If no context is specified then this context is used by default.
-
-*Example*
-
-Request:
+[[painless-execute-runtime-context]]
```

Comment:
Should this be at the top or lower? I feel like these are less important than the default context at least. Or, if not less important, less common.

Reply:
I'm going to remove the **Examples** header and instead bump the different context sections up a level to make them more prominent in the side navigation.